src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -1949,12 +1949,13 @@ class ObjectGenerator : public Generator<bsoncxx::document::value> {
     OnDuplicatedKeys _onDuplicatedKeys;
 };
 
-
 class IncGenerator : public Generator<int64_t> {
 public:
     IncGenerator(const Node& node, GeneratorArgs generatorArgs)
         : _step{node["step"].maybe<int64_t>().value_or(1)} {
-        _counter = node["start"].maybe<int64_t>().value_or(1) +
+       
+        int64_t start = node["start"] ? intGenerator(node["start"], generatorArgs)->evaluate() : 1;
```

Comment:
Can you add some tests and examples?

Generally we add tests [here](https://github.com/mongodb/genny/blob/master/src/value_generators/test/DocumentGeneratorTestCases.yml#L3) and add document some examples [here](https://github.com/mongodb/genny/blob/master/src/workloads/docs/Generators.yml#L57).

Reply:
thanks for the reminder, will add tests and examples

Reply:
added